crates/node/src/command.rs

```diff
@@ -52,69 +52,133 @@ impl SubstrateCli for Cli {
     }
 }
 
-/// Parse and run command line arguments
-pub fn run() -> sc_cli::Result<()> {
-    let mut cli = Cli::from_args();
-
-    // alias madara_path <> base_path
-    // TODO also alias tmp (tmp generates random base_paths that are not specified within
-    // the command)
-    let madara_path = match (cli.run.madara_path.clone(), cli.run.run_cmd.shared_params.base_path.clone()) {
+fn set_madara_path(cli: &mut Cli) {
+    match (cli.run.madara_path.clone(), cli.run.run_cmd.shared_params.base_path.clone()) {
         (Some(madara_path), _) => {
-            cli.run.run_cmd.shared_params.base_path = Some(madara_path.clone());
-            madara_path.to_str().unwrap().to_string()
+            cli.run.run_cmd.shared_params.base_path = Some(madara_path);
         }
         (_, Some(base_path)) => {
-            cli.run.madara_path = Some(base_path.clone());
-            base_path.to_str().unwrap().to_string()
+            cli.run.madara_path = Some(base_path);
```

Comment:
And if as you told "madara_path is always set" because it has a default value of `~/.madara`, we won't ever enter this arm.

Just do:
```rust
cli.run.run_cmd.shared_params.base_path = cli.run.madara_path.clone();
```